index.js

```diff
@@ -9,39 +8,22 @@ module.exports = function(jadeOptions, locals) {
     var protocol = require('protocol');
     var options = extend({}, jadeOptions || {});
 
-    protocol.interceptBufferProtocol('file', function(request, callback) {
-      var file = request.url.substr(7);
-      var content = null;
+    protocol.interceptProtocol('file', function(request) {
+        if(request) {
+          var file = request.url.substr(8);
+          if(file.endsWith('.jade')) {
+            var compiled = jade.compileFile(file, jadeOptions)(locals);
 
-      // See if file actually exists
-      try {
-        content = fs.readFileSync(file, 'utf8');
-      } catch (e) {
-        // See here for error numbers:
-        // https://code.google.com/p/chromium/codesearch#chromium/src/net/base/net_error_list.h
-       if (e.code === 'ENOENT') {
-         // NET_ERROR(FILE_NOT_FOUND, -6)
-         callback(6);
-       }
-
-       // All other possible errors return a generic failure
-       // NET_ERROR(FAILED, -2)
-       callback(2);
-      }
-
-      if (path.extname(file) === '.jade') {
-        var compiled = jade.compileFile(file, jadeOptions)(locals);
-
-        callback({data: new Buffer(compiled), mimeType:'text/html'});
-      } else {
-        callback({data: new Buffer(content)});
-      }
-    }, function (error, scheme) {
-      if (!error) {
-        console.log('jade interceptor registered successfully');
-      } else {
-        console.error('Jade interceptor failed:', error);
-      }
+            return new protocol.RequestStringJob({
```

Comment:
This is not compatible with the new [`protocol` API](https://github.com/atom/electron/blob/master/docs/api/protocol.md). The result should be provided using the callback function.